sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/OnEventListener.java

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+@FunctionalInterface
+public interface OnEventListener {
```

Comment:
Why do we need two listener interfaces?

Reply:
This was just kept around for the the other alternative to the original proposal. Will delete it.